c++/src/kj/async.c++

```diff
@@ -2575,6 +2575,17 @@ void WeakFulfillerBase::disposeImpl(void* pointer) const {
     delete this;
   } else {
     if (inner->isWaiting()) {
+      // Let's find out if there's an exception being thrown. If so, we'll use it to reject the
+      // promise.
+#if !KJ_NO_EXCEPTIONS
+      InFlightExceptionIterator iter;
+      KJ_IF_MAYBE(e, iter.next()) {
+        auto copy = kj::cp(*e);
+        copy.truncateCommonTrace();
+        inner->reject(kj::mv(copy));
+      }
+#endif
+      // Darn, use a generic exception.
       inner->reject(kj::Exception(kj::Exception::Type::FAILED, __FILE__, __LINE__,
```

Comment:
Not a big deal since this is a pretty small allocation and it looks like duplicate `reject` calls are noops, but just FYI in case you do care this is potentially calling `reject` a second time.

Reply:
Whoops, yes, that's a bug -- and the next commit actually makes this considerably more expensive than an allocation by adding a backtrace. Thanks.